core/src/services/aliyun_drive/backend.rs

```diff
@@ -0,0 +1,457 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use std::fmt::Debug;
+use std::fmt::Formatter;
+use std::sync::Arc;
+
+use bytes::Buf;
+use chrono::Utc;
+use log::debug;
+use serde::Deserialize;
+use tokio::sync::Mutex;
+
+use super::core::*;
+use super::lister::AliyunDriveLister;
+use super::lister::AliyunDriveParent;
+use super::reader::AliyunDriveReader;
+use super::writer::{AliyunDriveWriter, AliyunDriveWriters};
+use crate::raw::*;
+use crate::*;
+
+/// Aliyun Drive services support.
+#[derive(Default, Deserialize)]
+#[serde(default)]
+#[non_exhaustive]
+pub struct AliyunDriveConfig {
+    /// root of this backend.
+    ///
+    /// All operations will happen under this root.
+    ///
+    /// default to `/` if not set.
+    pub root: Option<String>,
+    /// client_id of this backend.
+    ///
+    /// required.
+    pub client_id: String,
+    /// client_secret of this backend.
+    ///
+    /// required.
+    pub client_secret: String,
+    /// refresh_token of this backend.
+    ///
+    /// required.
+    pub refresh_token: String,
+    /// drive_type of this backend.
+    ///
+    /// All operations will happen under this type of drive.
+    ///
+    /// default to DriveType::Default if not set.
+    pub drive_type: DriveType,
```

Comment:
It's better to use `String` in config instead of an new type to make our config parser much easier to implement.